spring-kafka/src/main/java/com/baeldung/partitioningstrategy/KafkaProducer.java

```diff
@@ -0,0 +1,26 @@
+package com.baeldung.partitioningstrategy;
+
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Service;
+
+@Service
+public class KafkaProducer {
```

Comment:
What is the purpose of this wrapper? Can we remove it and use the KafkaTemplate directly? I'm worried that in the article we refer to kafkaProducer.send rather than the standard  kafkaTemplate.send and so it might not be clear that we are using standard Kafka library features

Reply:
At first, I was using it to do some object transforming. But I find it defeats the purpose and will make the example complicated.  I agree with you that should use the standard kafkaTemplate instead. Updated.